virttest/utils_disk.py

```diff
@@ -807,8 +808,12 @@ def set_drive_letter(session, did, partition_no=1):
         if re.search("Reserved", line, re.I | re.M):
             partition_no += int(line.split()[1])
     assign_letter_cmd = ' echo select partition %s ; echo assign '
+    if target_letter:
+        assign_letter_cmd += 'letter=%s' % target_letter
     assign_letter_cmd = _wrap_windows_cmd(assign_letter_cmd)
     session.cmd(assign_letter_cmd % (did, partition_no))
+    if target_letter:
+        return target_letter
```

Comment:
Can the assignment fail in some cases? for example, specified with `E` but got other, if so, you have to let it run into the end to get the letter properly.

Reply:
I think we could raise an error if got another letter that is not you want instead of returning it since the purpose of the function is to get the right letter you want.
BTW, since we use the windows command` assign letter=` to set letter that we believe this command validity.
So did we need to check whether it works correctly? @luckyh

Reply:
Hi, @XueqiangWei 
About https://github.com/avocado-framework/avocado-vt/pull/2753#discussion_r486976573 and https://github.com/avocado-framework/avocado-vt/pull/2753#discussion_r486994290, do you have any comment?
Thanks.

Reply:
@YongxueHong the assigned target drive may be invalid or in use, if then session.cmd will raise the error. 
return None in the original codes, is because we want to handle the error when invoking it.
And I found you write a new function \"get_drive_letters\", I think it's OK to check the target letter.

By the way, for @menli820 's issue (The drive letter is changed from I to E), I think it should work well if she using the new function utils_disk.configure_empty_windows_disk.

Reply:
> @YongxueHong the assigned target drive may be invalid or in use, if then session.cmd will raise the error.
> return None in the original codes, is because we want to handle the error when invoking it.
> And I found you write a new function "get_drive_letters", I think it's OK to check the target letter.

Thanks for your suggestion, Hi @menli820  suggest to update by @luckyh and @XueqiangWei comments.

> 
> By the way, for @menli820 's issue (The drive letter is changed from I to E), I think it should work well if she using the new 
function utils_disk.configure_empty_windows_disk.

Hi @XueqiangWei,
I suggest changing the disk letter instead of format the disk again since do not destroy the metadata of original disk and the data.
Thanks.

Reply:
Thanks all, will update it.

Reply:
@YongxueHong Yes, I agree with you, I mean using \"utils_disk.configure_empty_windows_disk\" instead of \"utils_misc.format_windows_disk" in her case. It's old function in the case "iozone_windows". I think just using "utils_disk.configure_empty_windows_disk" should avoid her issue. Let's wait her feedback. Thanks.

Reply:
> @YongxueHong Yes, I agree with you, I mean using "utils_disk.configure_empty_windows_disk" instead of "utils_misc.format_windows_disk" in her case. It's old function in the case "iozone_windows". I think just using "utils_disk.configure_empty_windows_disk" should avoid her issue. Let's wait her feedback. Thanks.

Hi @XueqiangWei, sorry, my understand mistake, got it.